src/v/cluster/partition.h

```diff
@@ -62,7 +67,7 @@ class partition {
     ~partition();
 
     raft::group_id group() const { return _raft->group(); }
-    ss::future<> start();
+    ss::future<> start(std::optional<topic_configuration>);
```

Comment:
i think it is a bad idea to pass the topic configuration into partition, seems like an abstraction leak

Reply:
Also doesn't this mean that changes to topic configuration aren't captured until the partition is moved? In this specific case if the number of partitions are increased the coordinator doesn't know that.

Reply:
right, this is only the "initial" value (used when the stm is created for the first time). Subsequent count is applied from the log state. Repartitioning requires an explicit  `distributed_kv_stm::repartition(new_partition_count)` but currently that is not exposed in the transform client, we could probably add an admin_end point if we really need it.